api/v1alpha3/awsmachine_types.go

```diff
@@ -127,6 +127,10 @@ type AWSMachineSpec struct {
 	// SpotMarketOptions allows users to configure instances to be run using AWS Spot instances.
 	// +optional
 	SpotMarketOptions *SpotMarketOptions `json:"spotMarketOptions,omitempty"`
+
+	// Tenancy indicates if instance should run on shared or single-tenant hardware.
+	// +optional
+	Tenancy string `json:"tenancy,omitempty"`
```

Comment:
Does list have a set of predefined possibilities? Could use some kubebuilder validation to ensure the correct values are specified `// +kubebuilder:validation:Enum:=foo,bar,baz`

Reply:
`default | dedicated | host` are the options

Reply:
could also validate only if tenancy != "" or make it a pointer and check tenancy != nil so existing resources remain untouched. Any option seems reasonable to me.

Reply:
I added validation and defaulting webhook